dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -182,6 +187,90 @@ private static void printStaticLinkageErrors(
     return jarPaths.build();
   }
 
+  /**
+   * Finds linkage errors in the input classpath and generates a static linkage check report.
+   */
+  StaticLinkageCheckReport findLinkageErrors() {
+    ImmutableList<Path> jarFilePaths = classDumper.getInputClasspath();
+
+    ImmutableMap.Builder<Path, SymbolTable> jarToSymbols = ImmutableMap.builder();
+    for (Path jarPath : jarFilePaths) {
+      jarToSymbols.put(jarPath, scanExternalSymbolTable(jarPath));
+    }
+
+    // Validate linkage error of each reference
+    ImmutableList.Builder<JarLinkageReport> jarLinkageReports = ImmutableList.builder();
+    for (Map.Entry<Path, SymbolTable> entry : jarToSymbols.build().entrySet()) {
+      jarLinkageReports.add(findInvalidReferences(entry.getKey(), entry.getValue()));
+    }
+
+    if (reportOnlyReachable) {
+      // TODO: Optionally, report errors only reachable from entry point classes
```

Comment:
Maybe warn that this is unimplemented for now?

Reply:
throw new UnsupportedOperationException()?